content/tutorial/tutorial.md

```diff
@@ -513,63 +519,64 @@ class Board extends React.Component {
 }
 ```
 
-**[View the full code at this point](https://codepen.io/gaearon/pen/ybbQJX?editors=0010)**
+**[अभी तक का पूरा कोड देखें](https://codepen.io/gaearon/pen/ybbQJX?editors=0010)**
 
-After these changes, we're again able to click on the Squares to fill them, the same as we had before. However, now the state is stored in the Board component instead of the individual Square components. When the Board's state changes, the Square components re-render automatically. Keeping the state of all squares in the Board component will allow it to determine the winner in the future.
+इन परिवर्तनों के बाद, यह चंगेस करने के बाद, हम पहले के जैसे Squares को क्लिक कर पा रहे। हालाँकि, अब state को अलग-अलग Square कौम्पोनॅन्ट के बजाय Board कौम्पोनॅन्ट में संग्रहीत किया जाता है। जब Board का state बदल जाता है, तो Square कौम्पोनॅन्ट स्वचालित रूप से फिर से render करते हैं। Board कौम्पोनॅन्ट में सभी स्क्वायरस की स्थिति को ध्यान में रखते हुए यह भविष्य में विजेता का निर्धारण करने की अनुमति देगा।
 
-Since the Square components no longer maintain state, the Square components receive values from the Board component and inform the Board component when they're clicked. In React terms, the Square components are now **controlled components**. The Board has full control over them.
+चूंकि Square कौम्पोनॅन्ट अब state को बनाए नहीं रखते हैं, इसलिए Square कौम्पोनॅन्ट Board कौम्पोनॅन्ट से वैल्यू प्राप्त करते हैं और क्लिक किए जाने पर Board कौम्पोनॅन्ट को सूचित करते हैं। React के संदर्भ में, Sqaure कौम्पोनॅन्ट अब **controlled components** हैं। Board का उन पर पूरा नियंत्रण है।
 
-Note how in `handleClick`, we call `.slice()` to create a copy of the `squares` array to modify instead of modifying the existing array. We will explain why we create a copy of the `squares` array in the next section.
+ध्यान दें कि कैसे `handleClick` में, हम मौजूदा array को संशोधित करने के बजाय संशोधित करने के लिए `Square` array की एक प्रति बनाने के लिए `.slice()` कहते हैं। हम बताएंगे कि हम अगले भाग में `Square` array की एक कॉपी क्यों बनाते हैं।
 
-### Why Immutability Is Important {#why-immutability-is-important}
+### अचल स्थिति होना क्यों महत्वपूर्ण है {#why-immutability-is-important}
 
-In the previous code example, we suggested that you use the `.slice()` operator to create a copy of the `squares` array to modify instead of modifying the existing array. We'll now discuss immutability and why immutability is important to learn.
+पिछले कोड उदाहरण में, हमने सुझाव दिया कि आप मौजूदा सरणी को संशोधित करने के बजाय संशोधित करने के लिए `Square` array की एक प्रति बनाने के लिए `.slice()` ऑपरेटर का उपयोग करें। अब हम अपरिवर्तनीयता पर चर्चा करेंगे और सीखने के लिए अपरिवर्तनशीलता क्यों महत्वपूर्ण है।
```

Comment:
सरणी ?

Reply:
revisit as well

Reply:
> सरणी ?

array -> we should not translate, changing it

Reply:
```
पिछले कोड उदाहरण में, हमने सुझाव दिया कि `.slice()` ऑपरेटर का इस्तेमाल करके `squares` array की प्रति बनाये और मौजूदा array का संसोधन करने के बजाये नए array का संसोधन करें 
```

How about this?